bittensor/utils/registration/async_registration.py

```diff
@@ -0,0 +1,536 @@
+    """
+    if not num_processes:
+        # get the number of allowed processes for this process
+        num_processes = min(1, get_cpu_count())
+
+    if update_interval is None:
+        update_interval = 50_000
+
+    curr_block, curr_block_num, curr_diff = Solver.create_shared_memory()
+
+    solution = await _block_solver(
+        subtensor=subtensor,
+        wallet=wallet,
+        num_processes=num_processes,
+        netuid=netuid,
+        dev_id=None,
+        tpb=None,
+        update_interval=update_interval,
+        curr_block=curr_block,
+        curr_block_num=curr_block_num,
+        curr_diff=curr_diff,
+        n_samples=n_samples,
+        alpha_=alpha_,
+        output_in_place=output_in_place,
+        log_verbose=log_verbose,
+        cuda=False,
+    )
+
+    return solution
+
+
+async def create_pow_async(
+    subtensor: "AsyncSubtensor",
+    wallet: "Wallet",
+    netuid: int,
+    output_in_place: bool = True,
+    cuda: bool = False,
+    dev_id: Union[list[int], int] = 0,
+    tpb: int = 256,
+    num_processes: int = None,
+    update_interval: int = None,
+    log_verbose: bool = False,
+) -> Optional[dict[str, Any]]:
+    """
+    Creates a proof of work for the given subtensor and wallet.
+
+    Args:
+        subtensor (bittensor.core.async_subtensor.AsyncSubtensor): The subtensor object to use to get the block number, difficulty, and block hash.
+        wallet (bittensor_wallet.Wallet): The wallet to create a proof of work for.
+        netuid (int): The netuid for the subnet to create a proof of work for.
+        output_in_place (bool): If true, prints the progress of the proof of work to the console in-place. Meaning the progress is printed on the same lines.
+        cuda (bool): If true, uses CUDA to solve the proof of work.
+        dev_id (Union[list[int], int]): The CUDA device id(s) to use. If cuda is true and dev_id is a list, then multiple CUDA devices will be used to solve the proof of work.
+        tpb (int): The number of threads per block to use when solving the proof of work. Should be a multiple of 32.
+        num_processes (int): The number of processes to use when solving the proof of work. If None, then the number of processes is equal to the number of CPU cores.
+        update_interval (int): The number of nonces to run before checking for a new block.
+        log_verbose (bool): If true, prints the progress of the proof of work more verbosely.
+
+    Returns:
+        The proof of work solution or None if the wallet is already registered or there is a different error.
+
+    Raises:
+        ValueError: If the subnet does not exist.
+    """
+    if netuid != -1:
+        if not await subtensor.subnet_exists(netuid=netuid):
+            raise ValueError(f"Subnet {netuid} does not exist")
+
```

Comment:
```suggestion
    solution: Optional[POWSolution]
```

Don't redeclare the types later